packages/data-provider/src/utils/auditUtils.ts

```diff
@@ -0,0 +1,51 @@
+import * as _ from 'lodash-es';
+
+import { AUDIT_ACTION, AuditAction, AuditDataRepository, AuditDataResponse } from './types.js';
+
+/**
+ * Returns `true` if input value matches with a valid Audit Event type.
+ * Otherwise returns `false`
+ * @param {unknown} value
+ * @returns {boolean}
+ */
+export const isAuditEventValid = (value: unknown): boolean => {
+	return (
+		typeof value === 'string' &&
+		[AUDIT_ACTION.DELETE.toString(), AUDIT_ACTION.UPDATE.toString()].includes(value.toUpperCase())
+	);
+};
```

Comment:
These `toString()` calls are distracting, and only required because of the `includes` interface being strict to the array const types. Additionally, there is no guarantee that the AUDIT_ACTION constants are all upper case, so we should probably enforce that with a `toUpperCase` map.  

Consider instead:
```suggestion
const upperCaseAuditActions: string[] = Object.values(AUDIT_ACTION).map((value) => value.toUpperCase());
export const isAuditEventValid = (value: unknown): boolean => 
	typeof value === 'string' && upperCaseAuditActions.includes(value.toUpperCase());
```

I was going to recommend using a zod enum since we are writing our own matcher... but I can't figure out how to do case insensitive matching using zod enum.

Reply:
Updated this function to check if the audit event is valid using Zod safeParse.